sphinx/ext/autodoc/__init__.py

```diff
@@ -1075,43 +1086,27 @@ def format_args(self, **kwargs: Any) -> str:
     def document_members(self, all_members: bool = False) -> None:
         pass
 
-    def add_directive_header(self, sig: str) -> None:
+    def add_directive_header(self, sigs: List[str]) -> None:
```

Comment:
A documenter class is a public class for extensions. So this is a breaking change. I don't know there are really extensions that override this method. But we need a migration period for such change.

Reply:
Perhaps I'll change the caller to:
```
try:
    self.add_directive_header(sigs=Sig's)
except TypeError as e:
    self.add_directive_header(sig=sigs[0])
    emit_deprecation_warning()
```

Reply:
+1